### PR TITLE
CR-5108

### PR DESCRIPTION
**Pull request by elad-codefresh**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (ghost) [inline yarn.lock]: I guess the two yarn files got here by mistake

File: cmd/commands/git-source.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+		}
+	}
+
+	appDef := &runtime.AppDef{
+		Name: opts.gsName,
+		Type: application.AppTypeDirectory,
+		URL:  opts.gsCloneOpts.Repo,
+	}
+	if err := appDef.CreateApp(ctx, nil, opts.insCloneOpts, opts.runtimeName, store.Get().CFGitSourceType, nil); err != nil {
+		return err
```
Comment (ghost) [inline cmd/commands/git-source.go]:
```suggestion
		return fmt.Errorf("failed to create git-source application. Err: %w", err)
```

File: cmd/commands/git-source.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+				gsCloneOpts.Auth.Password = insCloneOpts.Auth.Password
+			}
+
+			insCloneOpts.Parse()
+			gsCloneOpts.Parse()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			err := RunCreateGitSource(ctx, &GitSourceCreateOptions{
```
Comment (ghost) [inline cmd/commands/git-source.go]: just return the RunCreateGitSource function call
the errors inside print good error messages to the user

File: cmd/commands/git-source.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+	log.G(ctx).Infof("done creating a new git-source: '%s'", opts.gsName)
+
+	return nil
+}
+
+func createDemoWorkflowTemplate(gsFs fs.FS, gsName, runtimeName string) error {
+	var err error
+
+	gsPath := gsFs.Join(apstore.Default.AppsDir, gsName, runtimeName)
```
Comment (ghost) [inline cmd/commands/git-source.go]: the gsPath can contain the "demo-wf-template.yaml" string here, and then there is no need to call `gsFs.join` again in line 199

File: cmd/commands/runtime.go [hunk 5]
```diff
@@ -258,7 +260,7 @@ func RunRuntimeInstall(ctx context.Context, opts *RuntimeInstallOptions) error {
 
 func NewRuntimeListCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "list ",
+		Use:     "list",
```
Comment (ghost) [inline cmd/commands/runtime.go]:
```suggestion
		Use:     "list [runtime_name]",
```

**Review by noam-codefresh — APPROVED** [review]

**Pull request merged**